presto-main/src/main/java/com/facebook/presto/memory/MemoryPool.java

```diff
@@ -178,17 +188,32 @@ public boolean tryReserve(QueryId queryId, String allocationTag, long bytes)
                 return false;
             }
             reservedBytes += bytes;
-            if (bytes != 0) {
-                queryMemoryReservations.merge(queryId, bytes, Long::sum);
-                updateTaggedMemoryAllocations(queryId, allocationTag, bytes);
+        }
+        if (bytes != 0) {
+            while (true) {
```

Comment:
i understand what this tries to do (basically either one of the following operation succeed, we are good). But it's a bit mind-twisting :). is this a common practice Java concurrency programming? ;)  cc @aweisberg , @arhimondr